handlers/analytics/log_processor.go

```diff
@@ -28,6 +29,7 @@ type LogProcessor struct {
 
 type LogDataEvent struct {
 	// Heartbeat event
+	Id                  *string `json:"id,omitempty"`
```

Comment:
re what happens when the field is not set, you're right that it will just be `nil`!